pkg/util/file_indexer.go

```diff
@@ -83,6 +85,60 @@ func resolveFilePath(directory string) (string, error) {
 	return directory, nil
 }
 
+// gitignoreFilePath gives the filepath of the .gitignore file in the context
+func gitIgnoreFilePath(directory string) (string, error) {
+	_, err := os.Stat(filepath.Join(directory))
+	if err != nil {
+		return "", err
+	}
+	return filepath.Join(directory, ".gitignore"), nil
+}
+
+// addOdoFileIndex adds odo-file-index.json to .gitignore
+func addOdoFileIndex(ignoreFile string) error {
+	var data []byte
+	file, err := os.OpenFile(ignoreFile, os.O_APPEND|os.O_RDWR, 0600)
+	if err != nil {
+		return errors.Wrap(err, "failed to open .gitignore file")
+	}
+	defer file.Close()
+
+	if data, err = ioutil.ReadFile(ignoreFile); err != nil {
+		return errors.Wrap(err, "failed reading data from .gitignore file")
+	}
+	// check whether .odo/odo-file-index.json is already in the .gitignore file
+	if !strings.Contains(string(data), ".odo/odo-file-index.json") {
+		if _, err := file.WriteString("\n" + ".odo/odo-file-index.json"); err != nil {
+			return errors.Wrap(err, "failed to Add odo-file-index.json to .gitignore file")
+		}
+	}
+	return nil
+}
+
+// Check .gitignore file exists or not
+func checkGitIgnoreFile(directory string) error {
+
+	ignoreFile, err := gitIgnoreFilePath(directory)
+	_, err = os.Stat(ignoreFile)
+	files, _ := ioutil.ReadDir(filepath.Join(directory))
+
+	if len(files) > 1 {
+		if err != nil {
+			file, err := os.OpenFile(ignoreFile, os.O_WRONLY|os.O_CREATE, 0600)
+			if err != nil {
+				return errors.Wrap(err, "failed to create .gitignore file")
+			}
+			file.Close()
+		}
+		err = addOdoFileIndex(ignoreFile)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
```

Comment:
IIUC, `checkGitIgnoreFile` is doing multiple things:
1. Checks if the `.gitignore` file is present.
2. If it's not present, it creates the file.
3. Adds entry for `.odo/odo-file-index.json` in the `.gitignore` file.

I think we should separate the 3rd step into a separate function of its own. I get it that we've created a separate function `addOdoFileIndex` but calling it within `checkGitIgnoreFile` still defies the purpose of this function. We could add it to `file_indexer.go` after the call to `checkGitIgnoreFile`.

Reply:
@kadel @girishramnani @mik-dass WDYT? Can you please address if you have any suggestion for this.

Reply:
@prietyc123 I feel @dharmit is right here. This function should just check for the existence of the .gitignore file.Only step 1 should be performed here.

Reply:
@dharmit +1

Reply:
@dharmit I have modified the content addressed by you :-)